### PR TITLE
Fix/peer manager keep reconnecting too quickly

### LINKED ISSUE #2413: PeerManager tries to connect to already connected nodes
PeerManager is responsible for establishing connections when the number of peers is not at max.
It has been reported by the Trinity team that when trying to find more connections we reconnect to already connected nodes.

### PR DESCRIPTION
**Pull request by asdacap**
Resolves #2413, #2191

## Changes:
- It seems that we keep reconnecting to peers potentially unnecessarily in peer manager.
- Attempt to reduce number of connection during peer discovery by adding disconnect delay.
- Large amount of `ClientQuitting`, can't figureout why, adding large timeout to it reduces the time it take to find peers.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [ ] No

**Comments about testing , should you have some** (optional)

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by LukaszRozmej** [comment]
Would it also fix: https://github.com/NethermindEth/nethermind/issues/2191 ?

**Comment by asdacap** [comment]
Well, it adds a minimum 30 second delay before retrying a peer, so probably I guess.

File: src/Nethermind/Nethermind.Network.Stats/NodeStatsLight.cs [hunk 1]
```diff
@@ -113,9 +121,34 @@ public void AddNodeStatsDisconnectEvent(DisconnectType disconnectType, Disconnec
...
+                    UpdateDelayConnectDeadline(delay, NodeStatsEventType.RemoteDisconnectDelay);
+                }
+            }
+
             Increment(NodeStatsEventType.Disconnect);
         }
 
+        private void UpdateDelayConnectDeadline(TimeSpan delay, NodeStatsEventType reason)
+        {
+            DateTime newDeadline = DateTime.Now + delay;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network.Stats/NodeStatsLight.cs:144]: DateTime.Now can have issues with daylight saving time? Would move to UtcTime or DateTimeOffset.

File: src/Nethermind/Nethermind.Network.Stats/Model/NodeStatsEventType.cs [hunk 2]
```diff
@@ -49,6 +51,11 @@ public enum NodeStatsEventType
         SyncFailed,
         SyncCompleted,
 
-        Disconnect
+        LocalDisconnectDelay,
+        RemoteDisconnectDelay,
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network.Stats/Model/NodeStatsEventType.cs:55]: Do we want to add counters to: https://github.com/NethermindEth/nethermind/blob/6c507f3e73307c527e72275419c6c1db14c344cd/src/Nethermind/Nethermind.Network/PeerManager.cs#L507-L519 ?

**Review by LukaszRozmej — APPROVED** [review]
Approve but move to something more reliable than `DateTime.Now`

File: src/Nethermind/Nethermind.Network.Stats/Model/NodeStatsEventType.cs (hunk 2 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Network.Stats/Model/NodeStatsEventType.cs:55]: Not really, the counters are too much IMO. And no decrement that I can find.

**Review by LukaszRozmej — APPROVED** [review]
Would still use `DateTimeOffset.UtcNow` but I think it doesn't change anything functionally?

**Pull request merged**